docs/csharp/whats-new/csharp-9.md

```diff
@@ -0,0 +1,233 @@
+
+Starting in C# 9.0, you can add the `static` modifier to lambda expressions or anonymous methods. That has the same effect as the `static` modifier on local functions: a static lambda or anonymous function cannot capture local variables or instance state. This prevents accidentally capturing other variables.
+
+Covariant return types enables a function that overrides a base class virtual function to change the return type to a type derived from the type declared in the base class. This can be useful for Records, and for other types that support virtual clone or factory methods.
+
+Next, you can use discards as parameters to lambda expressions. This convenience enables you to avoid naming the argument, and the compiler may be able to avoid using it. You use the `_` for any argument.
+
+Finally, you can now apply attributes to local functions. These are useful to add nullable attribute annotations to local functions.
+
+## Support for code generators
+
+Two final features support C# code generators. C# code generators are a component you can write that is similar to a roslyn analyzer or code fix. The difference is that code generators analyze code and write new source code files as part of the compilation process. A typical code generator searches code for attributes or other conventions.
+
+A code generator read attributes or other code elements using the Roslyn analysis APIs. From that information, it adds new code to the compilation. Source generators can only add code; they are not allowed to modify any existing code in the compilation.
+
+The two features added for code generators are extensions to ***partial method syntax***, and ***module initializers***. First, the changes to partial methods. Before C# 9.0, partial methods are `private` but cannot specify an access modifier, have a `void` return, and cannot have `out` parameters. These restrictions meant that if no method implementation is provided, the compiler removes all calls to the partial method. C# 9.0 removes these restrictions, but requires that partial method declarations have an implementation. Code generators can provide that implementation. To avoid introducing a breaking change, the compiler considers any partial method without an access modifier to follow the old rules. If the partial method includes the `private` access modifier, the new rules govern that partial method.
+
+The second new feature for code generators is ***module initializers***. These are methods that have the <xref:System.Runtime.CompilerServices.ModuleInitializer> attribute attached to them. These methods will be called by the runtime when the assembly loads. A module initializer method:
```

Comment:
The `<xref:System.Runtime.CompilerServices.ModuleInitializer>` is not yet available, we may want to remove the xref for now?

Reply:
My bad. It should be `ModuleInitializerAttribute` (for the class name)